src/Salt/SiteBundle/Controller/CommentsController.php

```diff
@@ -162,30 +162,70 @@ public function downvoteAction(Comment $comment, UserInterface $user)
-        $rows[] = implode(',', array_filter($headers));
-        $comment_data = $repo->findBy([$itemType => $itemId]);
+        $headers = ['Framework Name', 'Node Address', 'HumanCodingScheme', 'User', 'Organization', 'Comment', 'Created Date', 'Updated Date'];
+        $rows[] = implode(',', $headers);
+        $lsItemRepo = $this->getDoctrine()->getManager()->getRepository(LsItem::class);
+        switch ($itemType)
+        {
+            case 'document':
+            {
+                $comment_data = $repo->findBy([$itemType => $itemId]);
+                $child_comment_rows = $this->csvArray($comment_data, $itemType);
+                foreach ($child_comment_rows as $child_row)
+                {
+                    $rows[] = $child_row;
+                }
+                $lsDoc = $this->getDoctrine()->getManager()->getRepository(LsDoc::class)->findOneById($itemId);
+                $lsDocChilds = $lsItemRepo->findBylsDoc($lsDoc->getId());
+                if (!empty($lsDocChilds)){
+                    foreach ($lsDocChilds as $lsDocChild){
+                        $childIds[] = $lsDocChild->getId();
+                    }
+                }
+                array_filter($childIds);
+                break;
+            }
+            case 'item':
+            {
+                $lsItem = $lsItemRepo->findOneById($itemId);
+                $lsItemRepo->allChilds($childIds, $lsItem);
+                $childIds[] = $itemId;
+                 break;
+            }
+        }
+        $comment_data = $repo->findBy(['item' => $childIds]);
+        $child_comment_rows = $this->csvArray($comment_data, 'item');
+        if (!empty($child_comment_rows))
+        {
+            foreach ($child_comment_rows as $child_row)
+            {
+                $rows[] = $child_row;
+            }
+        }
+        $content = implode("\n", $rows);
+        $response = new Response($content);
+        $response->headers->set('content_type', 'text/csv');
+        return $response;
+    }
 
+    public function csvArray($comment_data, $itemType)
+    {
         foreach ($comment_data as $comment) {
             $comments=[
                 ($itemType === 'item') ? '"'.$comment->getItem()->getLsDoc()->getTitle().'"' : '"'.$comment->getDocument()->getTitle().'"',
-                $url,
+                ($itemType === 'item') ? $this->generateUrl('doc_tree_item_view', ['id' => $comment->getItem()->getId()], UrlGeneratorInterface :: ABSOLUTE_URL) : $this->generateUrl('doc_tree_view', ['slug' => $comment->getDocument()->getId()], UrlGeneratorInterface :: ABSOLUTE_URL),
                 ($itemType === 'item') ? '"'.$comment->getItem()->getHumanCodingScheme().'"' : null,
                 '"'.$comment->getUser()->getUsername().'"',
                 '"'.$comment->getUser()->getOrg()->getName().'"',
                 '"'.$comment->getContent().'"',
```

Comment:
Manually adding `"` can lead to errors as content could have unescaped characters or `"s`